monai/transforms/utils.py

```diff
@@ -368,6 +369,70 @@ def check_non_lazy_pending_ops(
         warnings.warn(msg)
 
 
+def map_and_generate_sampling_centers(
+    label: NdarrayOrTensor,
+    spatial_size: Sequence[int] | int,
+    num_samples: int,
+    label_spatial_shape: Sequence[int],
+    num_classes: int | None = None,
+    image: NdarrayOrTensor | None = None,
+    image_threshold: float = 0.0,
+    max_samples_per_class: int | None = None,
+    ratios: list[float | int] | None = None,
+    rand_state: np.random.RandomState | None = None,
+    allow_smaller: bool = False,
+    warn: bool = True,
+) -> tuple[tuple]:
+    """
+    Combine "map_classes_to_indices" and "generate_label_classes_crop_centers" functions, return crop center coordinates.
+    Call "map_classes_to_indices" if indices is None, pull the shape from the label or image (if label is None),
+    and then call "generate_label_classes_crop_centers".
```

Comment:
```suggestion
    This calls `map_classes_to_indices`  to get indices from `label`, gets the shape from `label_spatial_shape` is given otherwise from the labels, calls `generate_label_classes_crop_centers`, and returns its results.
```